### PR TITLE
Full path CSS and JS imports, Restructure sass mixins and globals 

### PR DESCRIPTION
**Pull request by mateoclarke**

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -39,8 +39,8 @@
     "validate-map-style": "node src/Map/style/scripts/validate.js",
     "build-map-style": "yarn build-map-icons; node src/Map/style/scripts/build.js",
     "build-map-icons": "spritezero public/mapboxstyle/sprite src/Map/style/icons/; spritezero --retina public/mapboxstyle/sprite@2x src/Map/style/icons/",
-    "build-css": "node-sass-chokidar src/ -o src/",
-    "watch-css": "yarn run build-css && node-sass-chokidar src/ -o src/ --watch --recursive",
+    "build-css": "node-sass-chokidar --include-path ./src src/ -o src/",
+    "watch-css": "yarn run build-css && node-sass-chokidar --include-path ./src src/ -o src/ --watch --recursive",
```
Comment (mateoclarke) [inline frontend/package.json]: @briaguya, Hey you cool with this? @ifsimicoded figured out a way to remove all the ugly relative paths for sass. She's looking into something similar for js. She wants to make sure this doesn't raise any red flags for you.

File: frontend/.env [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#do not add sensitive information before adding to .gitignore
```
Comment (mateoclarke) [inline frontend/.env]: giphy 18

File: frontend/.env [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+
+#full path imports
+NODE_PATH=src/
```
Comment (ifsimicoded) [inline frontend/.env]: @briaguya you ok with the .env file approach? We can also include this NODE_PATH definition in package.json in all the appropriate build commands. 
IF we go the .env approach, how do you feel about defining REACT_APP_AWS_ENDPOINT='http://localhost:5000/graphql there.

Comment (briaguya) [reply frontend/.env]: I'm cool with the .env file approach, I'll make sure everything about it works well with travis.

File: frontend/package.json (hunk 1 above)
Comment (briaguya) [reply frontend/package.json]: I like it, no red flags here

**Comment by mateoclarke** [comment]
Shit, I squashed and merged before the TravisCI finished running. Looks like an error in my merge conflict resolution. _please hold_

**Pull request merged**